### PR TITLE
[Feature][Timeseries] Benchmark Dashboard Integration

### PR DESCRIPTION
**Pull request by prateekdesai04**
*Description of changes:*
This PR incorporates benchmarking and visualization of AG Timeseries.
Benchmark on the Master Branch are ran sequentially as updating the same CloudFormation bench stack simultaneously can lead to errors and making two stacks will be overkill of resources.

These changes use the latest AG-Bench and not the PyPi release.
This PR works with the PR in AG Bench related to timeseries: https://github.com/autogluon/autogluon-bench/pull/64

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflow_scripts/run_benchmark.sh
Comment (tonyhoo) [inline .github/workflow_scripts/run_benchmark.sh:17]: What is the intention of passing the -dev-branch explicitly? What is the default

File: .github/workflow_scripts/run_benchmark.sh
Comment (tonyhoo) [inline .github/workflow_scripts/run_benchmark.sh:25]: Is it correct to assume that the `MODULE` here pertains to the categories of `tabular`, `multimodal`, and `timeseries`?

File: .github/workflows/benchmark-command.yml
Comment (tonyhoo) [inline .github/workflows/benchmark-command.yml:29]: What does the dev means here in terms of preset? Do we have such preset supported in AG and bench?

File: .github/workflows/benchmark-command.yml
Comment (tonyhoo) [inline .github/workflows/benchmark-command.yml:37]: same here, these value options are a bit confusing

**Comment by tonyhoo** [comment]
Thank you for your effort. Could you include a PR link demonstrating a successful timeseries run with dashboard link?

File: .github/workflows/benchmark-command.yml
Comment (prateekdesai04) [reply .github/workflows/benchmark-command.yml:29]: I think here it is the versions of AG.
Yes, we have such preset supported for timeseries.

File: .github/workflow_scripts/run_benchmark.sh
Comment (prateekdesai04) [reply .github/workflow_scripts/run_benchmark.sh:25]: Yes, the assumption is correct.

File: .github/workflows/benchmark-command.yml
Comment (shchur) [inline .github/workflows/benchmark-command.yml:47]: This might apply more to [`autogluon.bench`](https://github.com/autogluon/autogluon-bench/blob/master/sample_configs/amlb_configs/constraints.yaml), but should we make the constraint names more consistent with other AMLB constraints (e.g., `10m4c` for 10 minutes, 4 cores rather than `test101`)?

File: .github/workflows/benchmark-command.yml
Comment (prateekdesai04) [reply .github/workflows/benchmark-command.yml:29]: Yes

File: CI/bench/timeseries/amlb_user_dir/frameworks_template.yaml
Comment (prateekdesai04) [reply CI/bench/timeseries/amlb_user_dir/frameworks_template.yaml:13]: Yes we need it, using this template file, the repo and branch is overwritten with that of the PR repo and branch.

**Comment by prateekdesai04** [comment]
> Thank you for your effort. Could you include a PR link demonstrating a successful timeseries run with dashboard link?

https://github.com/prateekdesai04/autogluon/pull/30#issuecomment-1786683574
(this was put locally in order to test, sometimes the dashboard may change if a newer benchmark is run)

File: CI/bench/timeseries/amlb_user_dir/constraints.yaml
Comment (prateekdesai04) [reply CI/bench/timeseries/amlb_user_dir/constraints.yaml:3]: Yes, we need the constraint as this will be picked up by Bench running in our CI.
Basically the flow is as follows: we use our own custom configs to specify constraints, presets, frameworks and benchmark datasets, and for bench to pick this up we need to store all this in something called _amlb-user-dir_, then a cloud config file is generated which ensures the custom configs are picked up rather than the default AMLB configs.
If nothing is provided, AMLB Configs are picked up by default.

File: .github/workflows/benchmark-command.yml
Comment (prateekdesai04) [reply .github/workflows/benchmark-command.yml:47]: Yes, sure I can make this change!

File: .github/workflows/benchmark-command.yml
Comment (prateekdesai04) [reply .github/workflows/benchmark-command.yml:37]: I can make changes to keep a better naming convention

File: .github/workflows/benchmark-command.yml
Comment (tonyhoo) [reply .github/workflows/benchmark-command.yml:29]: I'd appreciate some clarification on the need for this new dev preset. Could you provide further information on why it is being introduced?

File: .github/workflow_scripts/run_benchmark.sh
Comment (tonyhoo) [reply .github/workflow_scripts/run_benchmark.sh:17]: Thanks for clarification

File: .github/workflows/benchmark-command.yml
Comment (prateekdesai04) [reply .github/workflows/benchmark-command.yml:29]: It's because AG-Bench uses this preset for timeseries - https://github.com/autogluon/autogluon-bench/blob/master/sample_configs/amlb_configs/frameworks_example.yaml
And hence adding dev will make it pick up the version specified in the frameworks.yaml file

File: CI/bench/timeseries/amlb_user_dir/constraints.yaml
Comment (shchur) [reply CI/bench/timeseries/amlb_user_dir/constraints.yaml]: I see, so [these configs](https://github.com/autogluon/autogluon-bench/blob/master/sample_configs/amlb_configs/constraints.yaml#L3) are available in the `autogluon.bench` repo, but they aren't automatically downloaded when we install bench with `pip install autogluon.bench`. So we need to store these files somewhere to use them, right?

**Review by shchur — APPROVED** [review]
Thanks for addressing the comments, @prateekdesai04!

You might also want to wait for @tonyhoo's approval since I don't have much context about the workings of this part of the codebase.

**Pull request closed without merging**
